### ISSUE TITLE
Activate method?

### ISSUE THREAD

**Issue by christianmagill** [StoutLogic/acf-builder#29]
Thanks for this awesome library!

I'm pretty sure I'll be using it on all my ACF projects moving forward.

Reading through the documentation, I wonder why you don't simply include an activate method for your FieldsBuilder that sets up the necessary hook?

**Comment by stevep** [comment]
Thanks @christianmagill!

I'm interested in seeing how people are using it, so if you or anyone has anything to share I'd love to see it.

Regarding an activate method, maybe you can give me an example of what you mean? Do you mean have a function that calls `acf_add_local_field_group` and passes itself it?

```php
class FieldsBuilder
{

    ...

    function activate() 
    {
        add_action('acf/init', function()  {
           acf_add_local_field_group($this->build());
        });
    }

}

**Comment by christianmagill** [comment]
Yes, that's what I meant! Just a quicker/simpler way to activate.

I started playing with the library last night so will keep you updated as other ideas come to mind.

**Comment by stevep** [comment]
My only hesitation with that is that it adds specific WordPress and ACF function calls to the library. I don't think the library currently does that anywhere else.

It also assumes that hook you want is `acf/init` which is probably the case, but I want to be sure. Like #30 I'm gonna think about this a bit more. But thanks for the suggestion and don't hesitate to send more.

**Comment by Levdbas** [comment]
In the light of lazy translations that this is probably not a good idea anymore since you want to build your full groups within the acf/init hooks nowayday.
